ts/features/bonus/cgn/components/otp/OtpCodeComponent.tsx

```diff
@@ -0,0 +1,245 @@
+import React, { useEffect, useRef, useState } from "react";
+import { Animated, Easing, StyleSheet, View, ViewStyle } from "react-native";
+import { Millisecond, Second } from "italia-ts-commons/lib/units";
+import { fromNullable } from "fp-ts/lib/Option";
+import { Otp } from "../../../../../../definitions/cgn/Otp";
+import { IOColors } from "../../../../../components/core/variables/IOColors";
+import { addEvery } from "../../../../../utils/strings";
+import { BaseTypography } from "../../../../../components/core/typography/BaseTypography";
+import customVariables from "../../../../../theme/variables";
+import { Label } from "../../../../../components/core/typography/Label";
+import IconFont from "../../../../../components/ui/IconFont";
+import TouchableDefaultOpacity from "../../../../../components/TouchableDefaultOpacity";
+import { clipboardSetStringWithFeedback } from "../../../../../utils/clipboard";
+import { isTestEnv } from "../../../../../utils/environment";
+import I18n from "../../../../../i18n";
+
+type ProgressConfig = {
+  startPercentage: number;
+  endPercentage: number;
+};
+
+type Props = {
+  otp: Otp;
+  progressBaseBgColor?: ViewStyle["backgroundColor"];
+  progressBgColor?: ViewStyle["backgroundColor"];
+  onEnd: () => void;
+  progressConfig?: ProgressConfig;
+};
+
+const styles = StyleSheet.create({
+  container: {
+    backgroundColor: IOColors.white
+  },
+  otpContainer: {
+    paddingHorizontal: customVariables.contentPadding,
+    paddingVertical: 4,
+    marginRight: 20,
+    marginLeft: 20,
+    marginTop: 4,
+    marginBottom: 4,
+    justifyContent: "center",
+    backgroundColor: IOColors.white,
+    flexDirection: "row",
+    shadowColor: "#00274e",
+    borderColor: "black",
+    borderRadius: 8,
+    shadowOffset: {
+      width: 0,
+      height: 5
+    },
+    shadowOpacity: 0.1,
+    shadowRadius: 10,
+    elevation: 2
+  },
+  progressBase: {
+    alignSelf: "center",
+    marginTop: 10,
+    marginBottom: 10,
+    backgroundColor: "#d9d9d9",
+    width: "90%",
+    height: 6,
+    margin: 4,
+    borderRadius: 4
+  },
+  progress: {
+    alignSelf: "flex-start",
+    backgroundColor: IOColors.blue,
+    width: "50%",
+    height: 6,
+    borderRadius: 4
+  },
+  remainingTimeContainer: {
+    alignSelf: "center",
+    marginTop: 2,
+    width: "90%"
+  },
+  optCode: { fontSize: 30, padding: 4, marginRight: 4, textAlign: "center" }
```

Comment:
To make this code actually Monospace we need to add fontFamily attribute

Looking at the invision it looks like it should be a `fontSize: 36` was it too big?

```suggestion
  optCode: { fontSize: 30, padding: 4, marginRight: 4, textAlign: "center", fontFamily: Platform.select({
                    android: "RobotoMono",
                    ios: "Roboto Mono"
                  } }
```

Reply:
I added `RobotoMono` font family (thanks, I missed).  `BaseTypography` takes care to keep the right font depending on platform.

I guess 36 it too much. I made some tests and in certain cases the code is shown in 2 lines. Maybe can we review in with designers?